log/sequencer.go

```diff
@@ -381,15 +381,17 @@ func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, gu
 	// where the tokens come from.
 	if numLeaves > 0 {
 		tokens := int(float64(numLeaves) * quotaIncreaseFactor())
-		glog.V(2).Infof("Replenishing %v tokens for tree %v (numLeaves = %v)", tokens, logID, leaves)
-		if err := s.qm.PutTokens(ctx, tokens, []quota.Spec{
+		specs := []quota.Spec{
 			{Group: quota.Tree, Kind: quota.Read, TreeID: logID},
 			{Group: quota.Tree, Kind: quota.Write, TreeID: logID},
 			{Group: quota.Global, Kind: quota.Read},
 			{Group: quota.Global, Kind: quota.Write},
-		}); err != nil {
+		}
+		glog.V(2).Infof("Replenishing %v tokens for tree %v (numLeaves = %v)", tokens, logID, leaves)
+		if err := s.qm.PutTokens(ctx, tokens, specs); err != nil {
 			glog.Warningf("Failed to replenish %v tokens for tree %v: %v", tokens, logID, err)
 		}
+		quota.Metrics.IncReplenished(tokens, specs)
```

Comment:
Yes. My reasoning for not differentiating errors is that this metric, as the other quota-related metrics in the PR, records the server's interactions with the quota system. The server did ask for replenishment, so not recording it makes it seem like it's misbehaving, when in fact it was the quota system that failed. The exact picture of how the quota system is doing has to come from within itself, as we can't account for some things externally (time-based replenishment, cfg changes, disabled quotas, etc). Plus, if we use the outside metrics to record the intended interactions, we can see how well the count "closes" from the server's POV (ie, if tokens spent - tokens replenished = 0).

Let me know if that makes sense.